### ISSUE TITLE
Callback when particle hit's collision

### ISSUE THREAD

**Issue by ghost** [godotengine/godot-proposals#3863]
### Describe the project you are working on

Particles

### Describe the problem or limitation you are having in your project

I want a particle to make a sound when it hits the floor, so i want it to be able to make a callback whenever it does that, with it's position so i can play the sound from the particles (example, big particle bricks hitting the floor from destroyed brick wall)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Makes it possible (afaik, not possible yet at all)

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

a simple node callback with position as an argument or so

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Not as far as i know..

### Is there a reason why this should be core and not an add-on in the asset library?

Needs to be core. (afaik)

**Comment by Calinou** [comment]
Duplicate of https://github.com/godotengine/godot-proposals/issues/3349.

**Comment by ccuntd** [comment]
that's just cpu particles

**Comment by ccuntd** [comment]
not gpu

**Comment by ccuntd** [comment]
thats not a duplicate, that when the particle dissapears, not when it hits a material

**Comment by ccuntd** [comment]
thats to dissapear on collision, not emit signal

**Comment by Calinou** [comment]
> that's to disappear on any collision/callback when it disappears, not to emit a signal on collision and continue living (like i want)

This is not technically possible to implement on GPUParticles, as they cannot communicate what's going on back to the CPU (without making things very slow at least). This is pending on the implementation of collision in CPUParticles, which is not planned for 4.0.

**Comment by ccuntd** [comment]
nice
